docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,141 @@
+
+
+### Reference types
+
+Swift types fall into two categories: value types and reference types. The same is true for Objective-C types, which makes importing types easy: structs are imported as value types and Objective-C classes are imported as Swift class types. The same is not true for C++. In typical C++ fashion, there is no clear idiom for defining reference types. All types in C++ are declared in the same way, but some types have reference semantics and others don’t. To be able to express reference types in Swift natively, users must annotate their reference types as such, which will tell the compiler to import them as Swift classes (which have the same semantics). 
+
+When importing a reference type, there are three main sub-categories that reference type can fall into: 1) **immortal** reference types are “intentionally leaked” and live for the duration of the program 2) **manually managed** reference types are constructed and destroyed manually by the programmer, these are unsafe reference types and should maybe be renamed accordingly 3) **automatically managed** reference types provide retain and release operations which are used to safely destroy a value of the type after its last use. 
+
+Reference types, their instances, methods on reference types, and other APIs that use reference types generally fit well into the existing Swift model and their use should be allowed without restriction. While it is possible to define unsafe APIs that use reference types, these APIs will not be any less safe than their Swift or C++ counterpart, so there is no reason to dis-allow them. Note: the only case where C++ APIs using reference types must be dis-allowed is when there is not one level of indirection provided (either a reference or pointer). In this case, the C++ API is not using the type as a reference types, and thus breaks the reference type definition (above).
+
+As per our goal’s specification, this method of importing reference types allows C++ interoperability to have a clear, native mapping for a common C++ API pattern that builds a more general purpose solution off of the pre-established reference type bridging from Objective-C. Additionally, Swift preserves the same safety properties as C++ while providing an even safer option for the common pattern where reference types have a retain and release operation.
+
+### Value types
+
+As mentioned above, swift types fall into two categories: value types and reference types. Value types are relatively simple in Swift, C, and Objective-C, however, the same is not true for C++ where value types may have complex lifetime operations that are used to express non-native concepts (to Swift) such as value types that own memory. This case of owned value types is specifically difficult to map on to Swift as it is novel to the language and does not always work well with the existing model for Swift value types that are trivially copyable.  
+
+#### Trivial types
```

Comment:
It may be helpful to call out that `std::array` is an example of a trivial types (whereas `std::vector` is an example of an owned type).